frontend/src/store/objectStore.ts

```diff
@@ -71,19 +71,22 @@ export const useObjectStore = defineStore('objectStore', () => {
         })).data;
 
         const uniqueIds = [...new Set(permResponse.map((x: { objectId: string }) => x.objectId))];
-        const objects = (await objectService.listObjects({ objId: uniqueIds, ...params })).data;
-        const metadataResponse = (await objectService.getMetadata(null, { objId: uniqueIds })).data;
-
-        objects.map(async (obj: any) => {
-          const metadata = metadataResponse.find((x: Metadata) => x.objectId === obj.id);
-          // TODO: Tags
-
-          if (metadata) {
-            obj.metadata = metadata;
-            obj.name = metadata.metadata.find((x: { key: string }) => x.key === 'name')?.value;
-          }
-        });
 
+        let objects = null;
+        if (uniqueIds.length) {
+          objects = (await objectService.listObjects({ objId: uniqueIds, ...params })).data;
+          const metadataResponse = (await objectService.getMetadata(null, { objId: uniqueIds })).data;
+
+          objects.map(async (obj: any) => {
```

Comment:
If we're not using the output of the map function (and we're not returning anything by the looks of the callback body), this should be a forEach instead.